src/table/src/TableRow.js

```diff
@@ -1,164 +1,149 @@
-     * The height of the row. Remember to add paddings when using "auto".
-     */
-    height: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
-      .isRequired,
-
-    /**
-     * Function that is called on click and enter/space keypress.
-     */
-    onSelect: PropTypes.func,
-
-    /**
-     * Function that is called on click and enter/space keypress.
-     */
-    onDeselect: PropTypes.func,
-
-    /**
-     * Makes the TableRow selectable.
-     */
-    isSelectable: PropTypes.bool,
-
-    /**
-     * Makes the TableRow selected.
-     */
-    isSelected: PropTypes.bool,
-
-    /**
-     * Manually set the TableRow to be highlighted.
-     */
-    isHighlighted: PropTypes.bool,
-
-    /**
-     * The intent of the alert.
-     */
-    intent: PropTypes.oneOf(['none', 'success', 'warning', 'danger'])
-      .isRequired,
-
-    /**
-     * The appearance of the table row. Default theme only support default.
-     */
-    appearance: PropTypes.string.isRequired,
-
-    /**
-     * Theme provided by ThemeProvider.
-     */
-    theme: PropTypes.object.isRequired,
-
-    /**
-     * Class name passed to the table row.
-     * Only use if you know what you are doing.
-     */
-    className: PropTypes.string
-  }
-
-  static defaultProps = {
-    intent: 'none',
-    appearance: 'default',
-    height: 48,
-    onSelect: () => {},
-    onDeselect: () => {},
-    onKeyPress: () => {}
-  }
-
-  handleClick = e => {
-    if (typeof this.props.onClick === 'function') {
-      this.props.onClick(e)
+const TableRow = memo(forwardRef((props, ref) => {
+  const {
+    className,
+    height = 48,
+    children,
+    intent = 'none',
+    appearance = 'default',
+    tabIndex = -1,
+
+    // Filter out
+    onClick,
+    onKeyPress = () => {},
+    onSelect = () => {},
+    onDeselect = () => {},
+
+    isHighlighted,
+    isSelectable,
+    isSelected,
+    ...rest
+  } = props
+
+  const theme = useTheme()
+
+  const [mainRef, setMainRef] = useState()
+
+  const handleClick = e => {
+    if (typeof onClick === 'function') {
+      onClick(e)
     }
 
-    if (this.props.isSelectable) {
-      if (this.props.isSelected) {
-        this.props.onDeselect()
+    if (isSelectable) {
+      if (isSelected) {
+        onDeselect()
       } else {
-        this.props.onSelect()
+        onSelect()
       }
     }
   }
 
-  handleKeyDown = e => {
-    if (this.props.isSelectable) {
+  const handleKeyDown = e => {
+    if (isSelectable) {
       const { key } = e
       if (key === 'Enter' || key === ' ') {
-        this.props.onSelect()
+        onSelect()
         e.preventDefault()
       } else if (key === 'ArrowUp' || key === 'ArrowDown') {
         try {
-          manageTableRowFocusInteraction(key, this.mainRef)
+          manageTableRowFocusInteraction(key, mainRef)
         } catch (_) {}
       } else if (key === 'Escape') {
-        if (this.mainRef) this.mainRef.blur()
+        if (mainRef && mainRef instanceof Node) mainRef.blur()
```

Comment:
Why do we need `instanceof Node` here?

Reply:
just to make sure that whatever is passed back as the ref is an element, without it, it was failing to `blur`